modules/core/src/main/java/org/apache/ignite/internal/processors/cache/query/GridCacheQueryAdapter.java

```diff
@@ -654,12 +655,22 @@ private boolean isSafeLossPolicy() {
                 if (prj != null || part != null)
                     return nodes(cctx, prj, part);
 
-                if (cctx.affinityNode())
+                if (cctx.affinityNode() && !cctx.topology().localPartitionMap().hasMovingPartitions())
                     return Collections.singletonList(cctx.localNode());
 
                 Collection<ClusterNode> affNodes = nodes(cctx, null, null);
 
-                return affNodes.isEmpty() ? affNodes : Collections.singletonList(F.rand(affNodes));
+                List<ClusterNode> nodes = new ArrayList<>(affNodes);
+
+                Collections.shuffle(nodes);
+
+                for (ClusterNode node : nodes) {
+                    if (!cctx.topology().partitions(node.id()).hasMovingPartitions()) {
```

Comment:
!cctx.topology().partitions(node.id()).hasMovingPartitions() is throwing Nullpointer exception at runtime for non-pationed Cache and the Cluster starts without any Cache and then Cline Node join and populate the Cache. cctx.topology().partitions(node.id()) is returning null

Reply:
Hi @sumanentc , do you encounter NullPointerException with running Ignite? Feel free to report an issue in https://issues.apache.org/jira/projects/IGNITE/issues

Reply:
Yes , I get the issue when the client node tries to fetch the Cache data from the Cluster.

Reply:
Still I suppose that reporting an issue in https://issues.apache.org/jira/projects/IGNITE/issues is a proper option. If you would like to contribute a fix it would be awesome. Unfortunately I will not have a time to fix this problem in near future, sorry for that.

Reply:
raised ticket for the same: https://issues.apache.org/jira/browse/IGNITE-13317

Reply:
Thank you!